ansible/roles/uan_interfaces/tasks/can.yml

```diff
@@ -0,0 +1,269 @@
+        - "vlan007"
+
+  when: (uan_user_access_cfg | upper) == "CAN"
+
+- name: Configure CHN interface
+  block:
+
+  - name: Get list of possible CHN interfaces
+    shell: "ip -o link list | cut -f 2 -d ':' | tr -d ' ' | egrep 'hsn'"
+    register: chn_device_list
+
+  - name: FAIL if no customer access interfaces are found
+    fail:
+      msg: "No Customer Access interfaces found"
+    when: chn_device_list.stdout_lines | length < 1
+
+  - name: Ensure list of possible CHN interfaces are UP
+    shell: "ip link set dev {{ chn_device_list.stdout_lines | first }} up"
+    register: cmd_out
+    failed_when: cmd_out.rc not in [0, 1]
+
+  - name: "List possible {{ uan_user_access_cfg | upper }} interfaces"
+    debug:
+      msg: "{{chn_device_list.stdout_lines }}"
+
+  - name: Set IPADDR1 entry for CHN
+    when: can_ip is defined
+    lineinfile:
+      dest: "/etc/sysconfig/network/ifcfg-{{ chn_device_list.stdout_lines | first }}"
+      regex: '^IPADDR1=(.*)$'
+      line: "IPADDR1={{ can_ip }}/{{ can_prefix }}"
+
+  - name: "Configure ifroute-{{ chn_device_list.stdout_lines | first }}"
+    template:
+      src: ifroute-customer_access.j2
+      dest: "/etc/sysconfig/network/ifroute-{{ chn_device_list.stdout_lines | first }}"
+      owner: root
+      mode: 0600
+
+  - name: Set list of interfaces to restart for CHN
+    set_fact:
+      uan_interface_restart_list: 
+        - "nmn0"
+        - "{{ chn_device_list.stdout_lines | first }}"
+
+  when: (uan_user_access_cfg | upper) == "CHN"
+
+- name: "ifdown {{ uan_interface_restart_list }} interfaces to set the interface config and routes"
+  shell: "wicked ifdown {{ item }}"
+  changed_when: result.rc == 0
+  failed_when: result.rc != 0
+  register: result
+  with_items: "{{ uan_interface_restart_list }}"
+
+- name: Wait 5 seconds for the interfaces to go down
+  wait_for:
+    timeout: 5
+
+- name: Restarting wicked-nanny
+  shell: "systemctl restart wickedd-nanny"
+  changed_when: result.rc == 0
+  failed_when: result.rc != 0
+  register: result
+
+- name: Wait 5 seconds for wickedd-nanny to restart
+  wait_for:
+    timeout: 5
+
+- name: Remove existing default route
+  command: "ip route delete default"
+  changed_when: result.rc != 0 or result.rc == 2
+  failed_when: result.rc != 0 and result.rc != 2
+  register: result
+
+- name: "ifup {{ uan_interface_restart_list }} interfaces to set the interface config and routes"
+  shell: "wicked ifup {{ item }}"
+  changed_when: result.rc == 0
+  failed_when: result.rc != 0
+  register: result
+  with_items: "{{ uan_interface_restart_list }}"
+
+- name: Wait 10 seconds for the interfaces to come up
+  wait_for:
+    timeout: 10
+
+- name: Check that we have the default route we are expecting
+  when: not uan_customer_default_route | bool
+  shell: "ip route | grep default | grep -q '{{ customer_access_gateway }}'"
+  failed_when: result.rc != 0
+  register: result
+
+- name: Check that the vlan007 interface has the expected IP
```

Comment:
Is it accurate to still say `vlan007` when the CHN is being configured?

Reply:
Fixed!